catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessor.java

```diff
@@ -0,0 +1,355 @@
+   *     coordinate was detected, or a collection of {@link LatLon} center points otherwise.
+   */
+  @Nullable
+  private LiteralSuggestion getMgrsSuggestion(String query) {
+    List<String> matches = coordinateStringsFromQuery(query);
+    if (matches.isEmpty()) {
+      LOGGER.debug("No valid MGRS strings were found or could be inferred");
+      return null;
+    }
+    StringBuilder nameBuilder = new StringBuilder("MGRS: ");
+    List<UsngCoordinate> uniqueMatches =
+        matches
+            .stream()
+            .distinct()
+            .map(this::parseMgrsString)
+            .filter(Objects::nonNull)
+            .peek(mgrs -> nameBuilder.append("[ ").append(mgrs.toMgrsString()).append(" ] "))
+            .collect(Collectors.toList());
+    if (uniqueMatches.size() == 1) {
+      UsngCoordinate mgrsCoord = uniqueMatches.get(0);
+      BoundingBox bb = translator.toBoundingBox(mgrsCoord);
+      return new LiteralSuggestion(LITERAL, nameBuilder.toString(), latLonFromBoundingBox(bb));
+    }
+    List<LatLon> geo =
+        uniqueMatches
+            .stream()
+            .map(translator::toLatLon)
+            .map(d -> from(d.getLat(), d.getLon()))
+            .collect(Collectors.toList());
+    return new LiteralSuggestion(LITERAL, nameBuilder.toString(), geo);
+  }
+
+  @Nullable
+  private UsngCoordinate parseMgrsString(String mgrs) {
+    try {
+      return translator.parseMgrsString(mgrs);
+    } catch (ParseException e) {
+      LOGGER.debug(format("Detected string [%s] was not valid MGRS", mgrs), e);
+      return null;
+    }
+  }
+
+  private static List<LatLon> latLonFromBoundingBox(BoundingBox bb) {
+    return ImmutableList.of(
+        from(bb.getSouth(), bb.getWest()),
+        from(bb.getNorth(), bb.getWest()),
+        from(bb.getNorth(), bb.getEast()),
+        from(bb.getSouth(), bb.getEast()));
+  }
+
+  private static List<String> coordinateStringsFromQuery(String query) {
+    final Matcher matcher = PATTERN_MGRS_ANYPART.matcher(query.concat(" "));
+    final List<String> mgrsStrings = new ArrayList<>();
+
+    StringBuilder mgrsBuilder = new StringBuilder();
+    MgrsPartType previousType = MgrsPartType.NONE;
+
+    while (matcher.find()) {
+      String mgrsPart = matcher.group();
+      MgrsPartType type = getPartType(mgrsPart);
+      switch (type) {
+        case ZONE:
+          handleZoneWithPreviousType(mgrsPart, previousType, mgrsBuilder, mgrsStrings);
+          break;
+        case HUNDRED_KM:
+          handleHundredKmWithPreviousType(mgrsPart, previousType, mgrsBuilder);
+          break;
+        case NUMERIC:
+          handleNumericWithPreviousType(mgrsPart, previousType, mgrsBuilder, mgrsStrings);
+          break;
+        case NONE:
+          throw new IllegalArgumentException(
+              format("The provided mgrs part [ %s ] must have a type", mgrsPart));
+      }
+      if (LOGGER.isDebugEnabled()) {
```

Comment:
❓ Is this check needed?

Reply:
Since I was passing a builder I thought it would be necessary. Will remove.